### PR TITLE
test: add tests for storage's functionalities

### PR DESCRIPTION
**Pull request by blaketingyu**
These tests ensure that the storage's loading and reading functionality are working as intended

Tests are implemented so that the recipes in the json files are exactly identical to the recipe objects created in the program

Let's,

* Update the equals method to make checking for identical recipes to be easier

### CODE REVIEW

**Review by yusufaine — APPROVED** [review]
Good job on covering the majority of the tests 🔥 . I'll try and contribute to fixing the UniqueRecipeList test once this is merged. 

LGTM 🌮

**Review by aidanaden — CHANGES_REQUESTED** [review]
Looks mostly good to merge! Just a few qns/suggestions and 1 small name suggestion (which may or may not be good im not sure tbh)

File: src/main/java/seedu/address/model/recipe/UniqueRecipeList.java [hunk 1]
```diff
@@ -119,7 +119,13 @@ public boolean equals(Object o) {
...
         }
         UniqueRecipeList other = (UniqueRecipeList) o;
-        return this.internalList.equals(other.internalList);
+        // return this.internalList.equals(other.internalList);
+        for (Recipe r : this.internalList) {
+            if (!other.internalList.contains(r)) {
+                return false;
+            }
+        }
+        return true;
```
[the remark this answers is not in the document]
Comment (blaketingyu) [reply src/main/java/seedu/address/model/recipe/UniqueRecipeList.java:128]: The equals method for recipes is unable to check 2 identical recipes as of now. The function here is just a suggestion and attempt by Yusuf to make it work 🌮. He'll be trying to make it work after we've merge this PR.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 2]
```diff
@@ -52,27 +55,36 @@ public static ReadOnlyRecipeBook getSampleRecipeBook() {
...
-    //  */
-    // public static List<Ingredient> getIngredientList(String... strings) {
-    //     return Arrays.stream(strings).map(Ingredient::new).collect(Collectors.toList());
-    // }
+    /**
+       * Returns an ingredient list containing the list of strings given.
+       * @param strings ingredients that are to the list of ingredients.
+       * @return the list of ingredients containing the newly added ingredients.
+       * @throws ParseException if any of the fields given in {@code strings} is invalid.
+       */
```
Comment (Oddcorner) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:63]: There should be an empty line between the description and tags in Javadocs.

File: src/test/data/JsonRecipeBookStorageTest/invalidAndValidRecipeBook.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
+  "_comment": "Invalid Steps field",
```
Comment (Oddcorner) [inline src/test/data/JsonRecipeBookStorageTest/invalidAndValidRecipeBook.json:2]: This comment suggests that the following recipe should have an invalid steps field, however, it doesn't appear to be the case. Could this be an error?

**Review by Oddcorner — COMMENTED** [review]
Great work on these tests! Just a few comments on Javadoc format. Otherwise LGTM! 👍

File: src/test/data/JsonRecipeBookStorageTest/invalidAndValidRecipeBook.json (hunk 3 above)
Comment (blaketingyu) [reply src/test/data/JsonRecipeBookStorageTest/invalidAndValidRecipeBook.json:2]: Thanks for the heads up Robin! It should be `Invalid Tags` instead.

**Review by aidanaden — APPROVED** [review]
LGTM! 🍝 🍝 🍝 

nice la lets gET THEM TESTS GOIN!!! 🔥

**Pull request merged**